src/containerapp/azext_containerapp/custom.py

```diff
@@ -854,12 +854,12 @@ def containerapp_up(cmd,
             _get_registry_from_app(app, source)  # if the app exists, get the registry
         _get_registry_details(cmd, app, source)  # fetch ACR creds from arguments registry arguments
 
-    app.create_acr_if_needed()
-
     if source:
-        app.run_acr_build(dockerfile, source, quiet=False, build_from_source=not _has_dockerfile(source, dockerfile))
+        used_default_container_registry = app.run_source_to_cloud_flow(source, dockerfile)
```

Comment:
Local variable 'used_default_container_registry' might be referenced before assignment

Reply:
I will add a new initialization of this variable to make sure it is always assigned properly.